### PR TITLE
Upgrade to Webpack v4 [WIP]

### PR DESCRIPTION
**Pull request by tlrobinson**
* [ ] fix production builds
* [ ] fix chunking

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -107,17 +112,35 @@ var config = (module.exports = {
       // NOTE @kdoh - 7/24/18
       // icepick 2.x is es6 by defalt, to maintain backwards compatability
       // with ie11 point to the minified version
-      icepick: __dirname + "/node_modules/icepick/icepick.min"
+      icepick: __dirname + "/node_modules/icepick/icepick.min",
     },
   },
 
+  // optimization: {
```
Comment (kdoh) [inline webpack.config.js]: Can we just remove this or will we want to try and do something with this later?

File: webpack.config.js [hunk 2]
```diff
@@ -256,20 +279,34 @@ if (NODE_ENV !== "production") {
...
   config.output.devtoolModuleFilenameTemplate = "[absolute-resource-path]";
   config.output.pathinfo = true;
+
+  config.mode = "development";
 } else {
   // this is required to ensure we don't minify Chevrotain token identifiers
   // https://github.com/SAP/chevrotain/tree/master/examples/parser/minification
   const tokens = allTokens.map(currTok => chevrotain.tokenName(currTok));
-  config.plugins.push(
+  // config.plugins.push(
```
Comment (kdoh) [inline webpack.config.js]: Same question about this r.e. comment out vs remove.

File: webpack.config.js (hunk 1 above)
Comment (tlrobinson) [reply webpack.config.js]: That reminds me I need to make sure the bundle size is approximately the same.

**Comment by tlrobinson** [comment]
Unfortunately it looks like the production build (e.x. `./bin/build`) is failing (which isn't tested by CI) so we need to hold off on merging this.

**Comment by tlrobinson** [comment]
```
Running 'webpack' with NODE_ENV=production assemble and minify frontend assets...
(node:2946) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead

<--- Last few GCs --->

[2946:0x102801000]   719236 ms: Mark-sweep 1300.7 (1498.6) -> 1300.3 (1499.6) MB, 8911.9 / 0.0 ms  allocation failure GC in old space requested
[2946:0x102801000]   729019 ms: Mark-sweep 1300.3 (1499.6) -> 1300.3 (1461.1) MB, 9781.7 / 0.0 ms  last resort GC in old space requested
[2946:0x102801000]   738975 ms: Mark-sweep 1300.3 (1461.1) -> 1300.3 (1458.1) MB, 9955.4 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x1293b7e25879 <JSObject>
    1: reduce_vars [0x12936a4022d1 <undefined>:~5969] [pc=0x14d3288b0fcb](this=0x129346bba0b9 <AST_VarDef map = 0x129324e1e089>,tw=0x129392a404a1 <TreeWalker map = 0x129324e1d061>,descend=0x12938a6849b9 <JSFunction (sfi = 0x1293dfbb0921)>)
    2: arguments adaptor frame: 3->2
    3: visit [0x12936a4022d1 <undefined>:~6001] [pc=0x14d3288a9705](this=0x129392a404a1 <TreeWalker map = 0x129324e1d061>...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 3: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 4: v8::internal::Factory::NewFixedArray(int, v8::internal::PretenureFlag) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 5: v8::internal::HashTable<v8::internal::SeededNumberDictionary, v8::internal::SeededNumberDictionaryShape>::EnsureCapacity(v8::internal::Handle<v8::internal::SeededNumberDictionary>, int, v8::internal::PretenureFlag) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 6: v8::internal::Dictionary<v8::internal::SeededNumberDictionary, v8::internal::SeededNumberDictionaryShape>::Add(v8::internal::Handle<v8::internal::SeededNumberDictionary>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyDetails, int*) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 7: v8::internal::(anonymous namespace)::DictionaryElementsAccessor::AddImpl(v8::internal::Handle<v8::internal::JSObject>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, unsigned int) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 8: v8::internal::JSObject::AddDataElement(v8::internal::Handle<v8::internal::JSObject>, unsigned int, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
 9: v8::internal::Runtime_SetProperty(int, v8::internal::Object**, v8::internal::Isolate*) [/Users/tlrobinson/.nvm/versions/node/v8.12.0/bin/node]
10: 0x14d3276042fd
11: 0x14d3288b0fcb
12: 0x14d32760535f
13: 0x14d3288a9705
14: 0x14d328dac1ac
./bin/build: line 24:  2946 Abort trap: 6           NODE_ENV=production ./node_modules/.bin/webpack --bail
```

**Comment by camsaul** [comment]
@paulrosenzweig @kdoh do we still want this?

**Comment by paulrosenzweig** [comment]
I think this is old enough that we should start from scratch instead.

**Comment by rlotun** [comment]
just linking as an FYI: https://github.com/metabase/metabase/issues/14803

**Pull request closed without merging**